packages/amplication-server/src/core/project/project.resolver.ts

```diff
@@ -92,9 +103,58 @@ export class ProjectResolver {
   )
   @InjectContextValue(InjectableOriginParameter.UserId, "data.user.connect.id")
   async commit(@Args() args: CreateCommitArgs): Promise<Commit | null> {
+    const projectId = args.data.project.connect.id;
+
+    await this.validateProject(projectId);
+
     return this.projectService.commit(args);
   }
 
+  async validateProject(projectId: string): Promise<void> {
+    const project = await this.projectService.findUnique({
+      where: { id: projectId },
+    });
+    const workspace = await this.workspaceService.getWorkspace({
+      where: { id: project.workspaceId },
```

Comment:
all the code and logic should be done in the service, and never in the resolver

Reply:
Done